### PR TITLE
Prettify k8s gen deployment-dir param configmap and use lowercase for configmaps

### PR DESCRIPTION
**Pull request by renuka-fernando**
## Purpose
- Prettify k8s gen deployment-dir param configmap - fix https://github.com/wso2/product-apim-tooling/issues/701
- use lowercase for configmaps - fix https://github.com/wso2/product-apim-tooling/issues/700

## Samples
```
$ tree DeploymentArtifacts_petstore
DeploymentArtifacts_petstore
├── api_crd.yaml
├── api_meta.yaml
├── certificates
└── params_cm.yaml

1 directory, 3 files
```

ConfigMap

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: swaggerpetstore-params
data:
  params.yaml: |
    environments:
        - name:
          configs:
              endpoints:
                  production:
                  sandbox:
              security:
                  enabled:
                  type:
                  username:
                  password:
              deploymentEnvironments:
                  - displayOnDevportal:
                    deploymentEnvironment:
                    deploymentVhost:
                  - displayOnDevportal:
                    deploymentEnvironment:  
                    deploymentVhost:
              certs:
                  - hostName:
                    alias:
                    path:
                  - hostName:
                    alias:
                    path:
              mutualSslCerts:
                  - tierName:
                    alias:
                    path:
                  - tierName:
                    alias:
                    path:
              policies:
                  - 
                  -  
    

```

### CODE REVIEW

File: import-export-cli/cmd/k8s/genDeploymentDir.go [hunk 1]
```diff
@@ -143,10 +146,11 @@ func executeGenDeploymentDirCmd() error {
...
 			}
-			apiCrd.Name = apiMetaData.Name
-			apiCrd.Spec.SwaggerConfigMapName = fmt.Sprintf("%v-cm", apiMetaData.Name)
-			apiCrd.Spec.ParamsValues = fmt.Sprintf("%v-params", apiMetaData.Name)
-			apiCrd.Spec.CertsValues = fmt.Sprintf("%v-certs", apiMetaData.Name)
+			apiName := utils2.GetValidK8sResourceName(apiMetaData.Name)
+			apiCrd.Name = apiName
+			apiCrd.Spec.SwaggerConfigMapName = fmt.Sprintf("%v-cm", apiName)
+			apiCrd.Spec.ParamsValues = fmt.Sprintf("%v-params", apiName)
+			apiCrd.Spec.CertsValues = fmt.Sprintf("%v-certs", apiName)
```
Comment (wasuradananjith) [inline import-export-cli/cmd/k8s/genDeploymentDir.go:153]: Noticed that, this code segment is getting repeated in the code 3 times. Shall we move this to a separate function and reuse it?

File: import-export-cli/cmd/k8s/genDeploymentDir.go [hunk 2]
```diff
@@ -163,10 +167,11 @@ func executeGenDeploymentDirCmd() error {
 			if errUnmarshal != nil {
 				utils.HandleErrorAndExit("Error unmarshal api configmap into struct ", errUnmarshal)
 			}
-			apiCrd.Name = apiMetaData.Name
-			apiCrd.Spec.SwaggerConfigMapName = fmt.Sprintf("%v-cm", apiMetaData.Name)
-			apiCrd.Spec.ParamsValues = fmt.Sprintf("%v-params", apiMetaData.Name)
-			apiCrd.Spec.CertsValues = fmt.Sprintf("%v-certs", apiMetaData.Name)
+			apiName := utils2.GetValidK8sResourceName(apiMetaData.Name)
```
Comment (wasuradananjith) [inline import-export-cli/cmd/k8s/genDeploymentDir.go:170]: There is **fmt.Println(fileName)** in line 161 of this file. Shall we remove it, if it a debug log?

**Review by pubudu538 — APPROVED** [review]

File: import-export-cli/cmd/k8s/genDeploymentDir.go (hunk 1 above)
Comment (pubudu538) [reply import-export-cli/cmd/k8s/genDeploymentDir.go:153]: +1

File: import-export-cli/cmd/k8s/genDeploymentDir.go (hunk 2 above)
Comment (Jayanie) [reply import-export-cli/cmd/k8s/genDeploymentDir.go:170]: Yes, Shall we remove that line?

**Review by Jayanie — APPROVED** [review]

**Review by uvindra — APPROVED** [review]

**Review by wasuradananjith — APPROVED** [review]

File: import-export-cli/cmd/k8s/genDeploymentDir.go (hunk 2 above)
Comment (renuka-fernando) [reply import-export-cli/cmd/k8s/genDeploymentDir.go]: fixed with https://github.com/wso2/product-apim-tooling/pull/703/commits/b01140f8cb4f3c34eca58b05da64c057839cfddd

File: import-export-cli/cmd/k8s/genDeploymentDir.go (hunk 1 above)
Comment (renuka-fernando) [reply import-export-cli/cmd/k8s/genDeploymentDir.go]: fixed with https://github.com/wso2/product-apim-tooling/pull/703/commits/b01140f8cb4f3c34eca58b05da64c057839cfddd

**Pull request merged**